optimum/exporters/openvino/__main__.py

```diff
@@ -263,6 +263,44 @@ class StoreAttr(object):
         **loading_kwargs,
     )
 
+    if (
+        library_name == "diffusers"
+        and ov_config
+        and ov_config.quantization_config
+        and "dataset" in ov_config.quantization_config
+    ):
+        import huggingface_hub
```

Comment:
Great to make it available in the CLI, but would prefer to not have it included in `main_export` (which should be reserved for the model loading + export steps) but directly in the quantizer (that could be called after `main_export` in https://github.com/huggingface/optimum-intel/blob/main/optimum/commands/export/openvino.py)

Reply:
In my opinion : 
* `optimum.exporters.openvino` should have everything export related
* `optimum.intel.openvino.quantization` should have everything quantization related (potentially calling export functions from `optimum.exporters.openvino` when the model to quantize is a `torch.nn.Module`), also should handle everything related to OVModel quantization (for example I think https://github.com/huggingface/optimum-intel/blob/086fae35729ce8e52ee2e473b8f048afb3b0839e/optimum/intel/openvino/modeling_decoder.py#L616-L646 should be moved to the `OVQuantizer`)
what do you think ?

Reply:
I agree with this point but we need a way to call both export and optimization in CLI.
>>should be moved to the OVQuantizer)
@echarlaix, this is what @nikita-savelyevv started working on according to our recent agreements with you. We will create a separate PR for that.

Reply:
yes do you think we can first merge the refactorization PR as adding more features (like support of hybrid quantization through the CLI) will make the refactorization even more complex

Reply:
In any case I don't think we should create an instance of `OVModel` inside `main_export` (which itself will call `main_export`), it also results in the model be loaded twice. It would be easier to have it there : https://github.com/huggingface/optimum-intel/blob/main/optimum/commands/export/openvino.py#in my opinion

Reply:
what about moving conversion of sd from export_main into separated function  and calling quantization after conversion when models saved on  disk happens if config provided?

similar way used for weights compression, but weights compression applied on single model and we do not need to have whole pipeline for data collection, that is why I think it is better to separate sd models conversion to know where pipeline conversion finished. additionally I think it may help in this refactoring as it may be used inside pipeline with more control

Reply:
I separated the  model conversion from quantization if a dataset is provided for SD model.